apps/web/src/hooks/useCreateWorkflowFromBlueprint.ts

```diff
@@ -0,0 +1,35 @@
+import slugify from 'slugify';
+import { useMutation, UseMutationOptions } from '@tanstack/react-query';
+import type { INotificationTemplate } from '@novu/shared';
+
+import { useCreateTemplateFromBlueprint } from '../api/hooks';
+import { getBlueprintTemplateById, getTemplateById } from '../api/notification-templates';
+import { TemplateCreationSourceEnum } from '../pages/templates/shared';
+import { getWorkflowBlueprintDetails } from '../utils';
+
+export const useCreateWorkflowFromBlueprint = (
+  options: UseMutationOptions<INotificationTemplate, any, { blueprintIdentifier: string }> = {}
+) => {
+  const { mutateAsync: createTemplateFromBlueprint } = useCreateTemplateFromBlueprint();
+
+  const { mutate: createWorkflowFromBlueprint, ...rest } = useMutation(async ({ blueprintIdentifier }) => {
+    const blueprintData = await getBlueprintTemplateById(blueprintIdentifier);
+    const { name: blueprintName } = getWorkflowBlueprintDetails(blueprintData.name);
```

Comment:
Get the blueprint by constant identifier. The blueprint name contains the "font awesome icons" in the name that's why we have to strip it.

Reply:
I think at some point we'll want to revisit using FontAwesome icons in the names and just use unicode... It's a double-whammy having to have special handling _and_ an additional icon library

Reply:
yes, we need to rethink this part ;)